yarn-project/circuits.js/src/structs/kernel/combined_accumulated_data.ts

```diff
@@ -157,24 +167,18 @@ export class PublicDataTransition {
 
 export class CombinedAccumulatedData {
   constructor(
-    public aggregationObject: AggregationObject,
-
+    public aggregationObject: AggregationObject, // Contains the aggregated proof of all previous kernel iterations
     public privateCallCount: Fr,
     public publicCallCount: Fr,
-
-    public newCommitments: Fr[],
-    public newNullifiers: Fr[],
-
-    public privateCallStack: Fr[],
-    public publicCallStack: Fr[],
-    public l1MsgStack: Fr[],
-
-    public newContracts: NewContractData[],
-
-    public optionallyRevealedData: OptionallyRevealedData[],
-
-    public stateTransitions: PublicDataTransition[],
-    public stateReads: PublicDataRead[],
+    public newCommitments: TupleOf<Fr, typeof KERNEL_NEW_COMMITMENTS_LENGTH>,
+    public newNullifiers: TupleOf<Fr, typeof KERNEL_NEW_NULLIFIERS_LENGTH>,
+    public privateCallStack: TupleOf<Fr, typeof KERNEL_PRIVATE_CALL_STACK_LENGTH>,
+    public publicCallStack: TupleOf<Fr, typeof KERNEL_PUBLIC_CALL_STACK_LENGTH>,
+    public l1MsgStack: TupleOf<Fr, typeof KERNEL_L1_MSG_STACK_LENGTH>,
+    public newContracts: TupleOf<NewContractData, typeof KERNEL_NEW_CONTRACTS_LENGTH>,
+    public optionallyRevealedData: TupleOf<OptionallyRevealedData, typeof KERNEL_OPTIONALLY_REVEALED_DATA_LENGTH>,
+    public stateTransitions: TupleOf<PublicDataTransition, typeof STATE_TRANSITIONS_LENGTH>,
+    public stateReads: TupleOf<PublicDataRead, typeof STATE_READS_LENGTH>,
```

Comment:
So are these tuples? Or arrays of tuples? Consider renaming `TupleOf`?

Reply:
They're tuples, why rename?

Reply:
My suggestion was to rename them if they are not tuples but arrays of tuples

Reply:
Renamed to Tuple